src/env.cc

```diff
@@ -181,6 +193,26 @@ void IsolateData::CreateProperties() {
         sizeof(#Provider) - 1).ToLocalChecked());
   NODE_ASYNC_PROVIDER_TYPES(V)
 #undef V
+
+  // Create all the hook type strings that will be passed to JS. Place them in
+  // an array so the array index matches the type id offset. This way the
+  // strings can be retrieved quickly.
+#define PROMISE_HOOK_TYPES(V)                                                 \
+  V(0, init)                                                                  \
```

Comment:
shouldn't we use the enum values here (e.g `PromiseHookType::kInit`,...) to ensure values map?

Reply:
Yep, will do. 👍